File: common/src/main/java/su/plo/voice/socket/NettyPacketUdp.java
package su.plo.voice.socket;

import io.netty.channel.socket.DatagramPacket;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import su.plo.voice.proto.packets.udp.PacketUdp;

@RequiredArgsConstructor
public final class NettyPacketUdp {

    @Getter
    private final DatagramPacket datagramPacket;
    @Getter
    private final byte[] packetData;
    @Getter
    private final PacketUdp packetUdp;
}

File: common/src/main/kotlin/su/plo/voice/socket/NettyPacketUdp.kt
package su.plo.voice.socket

import io.netty.channel.socket.DatagramPacket
import io.netty.util.ReferenceCounted
import su.plo.voice.proto.packets.udp.PacketUdp

data class NettyPacketUdp(
    val datagramPacket: DatagramPacket,
    val packetUdp: PacketUdp,
) : ReferenceCounted {
    override fun refCnt() = datagramPacket.refCnt()

    override fun retain() = apply { datagramPacket.retain() }

    override fun retain(increment: Int) = apply { datagramPacket.retain(increment) }

    override fun touch() = apply { datagramPacket.touch() }

    override fun touch(hint: Any?) = apply { datagramPacket.touch(hint) }

    override fun release() = datagramPacket.release()

    override fun release(decrement: Int) = datagramPacket.release(decrement)
}

